### PR TITLE
GH-108: [feat] Added profile preference section after creating account

### PR DESCRIPTION
**Pull request by NicolasChelico**
The following PR includes the addition of the Profile preference page which is the page a user is redirected to after successful account creation to initialize their preferences. 

- Test cases refactor are incoming
- UI is subject to change in next sprint for spreading out over two pages making it a multi step process to remove clutter. ** To keep in mind**

Screenshot 2024-11-09 at 3 25 05 PM

### CODE REVIEW

**Comment by KhalilGarali** [comment]
I think we should wait for @NicolasChelico and @walidoow agree on the models before merging this PR.

**Comment by KhalilGarali** [comment]
When I checkout to your branch to test this is what I saw:
image

I'm not sure if this is UI friendly /  good user experience. You can maybe change it like so: take off the blue button, make the logo clikable to choose the date and display the data as string in the field. So the field would be readonly.

Also there is a small bug when I tested: when you choose your date for the first time, you cannot change it back again. This keeps popping up:
image

Steps to repo:
- open date time picker
- choose for example November 7th 2024. click done.
- open date picker time again.

File: ClientApp/app/auth/profilePreferenceForm.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+              <TextInput
+                placeholder="First name"
+                onBlur={onBlur}
+                onChangeText={onChange}
+                value={value}
+              />
+            )}
+          />
+        </View>
+        {errors.firstName && (
```
[the remark this answers is not in the document]
Comment (KhalilGarali) [reply ClientApp/app/auth/profilePreferenceForm.tsx:73]: @JRB958

**Review by KhalilGarali — CHANGES_REQUESTED** [review]

File: ClientApp/app/auth/profilePreferenceForm.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (NicolasChelico) [reply ClientApp/app/auth/profilePreferenceForm.tsx:73]: I agrée with you but this is using prettier formatting. I can move to other way instead

**Review by KhalilGarali — APPROVED** [review]
Approved

**Review by JRB958 — APPROVED** [review]
changes look good

**Pull request merged**